core/src/main/scala/org/apache/spark/storage/DiskStore.scala

```diff
@@ -126,6 +127,12 @@ private[spark] class DiskStore(
     }
   }
 
+  def moveFileToBlock(sourceFile: File, blockSize: Long, targetBlockId: BlockId): Unit = {
+    blockSizes.put(targetBlockId, blockSize)
```

Comment:
I assume that the temp block size is not in `blockSizes`? `TempFileBasedBlockStoreUpdater.toBlockData` calls a `DiskStore` API with a temp block id, so I was wondering if all the temp block metadata is tracked by this class, or just its location.

Reply:
Yes you are right: `blockSizes` does not contain the block size of the block stored in the temporary file so the `toBlockData` would return an invalid value but because `TempFileBasedBlockStoreUpdater` was only used for DISK_ONLY storage levels where the replication is 1 this part was never executed. 

I started to correct it then realised a bit more restructuring of the `BlockStoreUpdater` source we can achieve more improvements (in performance) and the code readability could be improved too. I will add some more tests too.